src/Containers/Microsoft.NET.Build.Containers/Tasks/CreateNewImage.cs

```diff
@@ -120,9 +120,17 @@ internal async Task<bool> ExecuteAsync(CancellationToken cancellationToken)
         (string[] entrypoint, string[] cmd) = DetermineEntrypointAndCmd(baseImageEntrypoint: imageBuilder.BaseImageConfig.GetEntrypoint());
         imageBuilder.SetEntrypointAndCmd(entrypoint, cmd);
 
-        foreach (ITaskItem label in Labels)
+        if (GenerateLabels)
         {
-            imageBuilder.AddLabel(label.ItemSpec, label.GetMetadata("Value"));
+            foreach (ITaskItem label in Labels)
+            {
+                imageBuilder.AddLabel(label.ItemSpec, label.GetMetadata("Value"));
+            }
+
+            if (GenerateDigestLabel)
```

Comment:
I think this nesting makes sense but should we have an explicit error case for `GenerateLables=false` + `GenerateDigestLabel=true`? Right now it’d be silently ignored, right?

Reply:
It would be ignored, but I'm not sure it should be an _error_, necessarily. I'm honestly not sure that anyone is using these toggles, I'm just trying to model good behavior :D 

We don't warn for any of the other conditional label setting that happens in the targets, so I think I'd rather keep the behavior aligned - this label only looks different because we don't have the digest to write until we're inside the CreateNewImage Task.

Reply:
My general preference is to explicitly block "weird" behavior because

* It might be worse than "weird" if it's so weird we didn't test it, and
* We can always remove an error but can't add one.

But for this thing in particular I don't feel strongly about it, happy to go with you here.

Reply:
I added an informational message rather than a warning/error so that there would be some signal